[WIP] APPS-3024: Integrator refactor WirePluginManifestStrategy with key (#14)
spryker-sdk/integrator: README.md

composer require --dev spryker-sdk/integrator
```

## How it works

> 1. Every time when the integrator is run it uploads recipes from `spryker-sdk/integrator-recipes` and uses the master branch. They are uploaded into this folder: `vendor/spryker-sdk/integrator/data/recipes/`.

Reviewer: ```suggestion
1. Every time the integrator runs it downloads recipes from `spryker-sdk/integrator-recipes` and uses the master branch. The recipes are downloaded into this folder: `vendor/spryker-sdk/integrator/data/recipes/`.
```

> 2. It gathers the list of dependencies from `composer.json` and their uploaded versions from `composer.lock` of the project.

Reviewer: ```suggestion
2. It gathers the list of dependencies from `composer.json` and their installed versions from `composer.lock` of the project.
```

3. The integrator checks manifests (recipes) versions and collect manifests that should be applied.
4. All gathered manifest are applied to the project code.

> ## How to test the specific recipe using suite-nonsplit

Reviewer: This approach is ok for now but needs to be changed. We should also move Spryker only related documentation into confluence.

I already have an idea and will create a ticket for it.

1. We should specify module and get content of needed manifest of this module. To do it we need change code in `\SprykerSdk\Integrator\Manifest\ManifestReader::readManifests` method:
```php
    /**
     * @param array<\SprykerSdk\Integrator\Transfer\ModuleTransfer> $moduleTransfers
     *
     * @return array<string, array<string, array<string>>>
     */
    public function readManifests(array $moduleTransfers): array
    {
        return [
            'Spryker.AssetExternal' => json_decode(
                file_get_contents('vendor/spryker-sdk/integrator/data/recipes/integrator-recipes-master/AssetExternal/1.0.0/installer-manifest.json'),
                true,
            ),
        ];
    }
```
2. Open CLI:
```shell
docker/sdk cli
```
3. Run integrator:
```shell
vendor/bin/integrator
```

> 4. The changes from the source manifest file should be applied to target places and should appear in the `integrator.lock` file in the source project directory.

Reviewer: ```suggestion
4. Changes defined in the manifest file will be applied on the project level and will be logged in the `integrator.lock` file in the root directory of the project.
```

### Why should we do in this way?
The integrator gathers dependencies from the `composer.json` file and it means that no one spryker module is instaled directly. All spryker modules are installed in the `spryker/spryker` package. Because of it, we can not apply recipes to the specific spryker module.

> We can test the integrator behavior and recipes as they are in the **B2C** because it installs spryker modules directly (recipes should be in the master branch of `spryker-sdk/integrator-recipes`).

Reviewer: We should also more this to the internal Spryker documentation.
